examples/go/stream_benchmarker/benchmark/main.go

```diff
@@ -0,0 +1,417 @@
+		AcceptedFraming: []stellarstation.Framing{stellarstation.Framing_AX25, stellarstation.Framing_BITSTREAM},
+		SatelliteID:     satelliteID,
+		APIKey:          apiKey,
+		Endpoint:        endpoint,
+	}
+
+	var streamChannel = make(chan *stellarstation.ReceiveTelemetryResponse)
+
+	ss, err := benchmark.OpenSatelliteStream(satelliteStreamOptions, streamChannel)
+	if err != nil {
+		log.Fatal(err)
+
+	}
+	defer ss.Close()
+
+	passEndedChan := make(chan bool)
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+	reportingTicker := time.NewTicker(interval)
+	defer reportingTicker.Stop()
+
+	log.Println("Listening for messages")
+	var ps passSummary
+	var md metricsData
+	var passSummaries []passSummary
+	var passStarted = false
+	var done = false
+	var printedHeader = false
+	for !done {
+		select {
+		case streamResponse := <-streamChannel:
+			message := streamResponse.Telemetry
+			if len(message.Data) != 0 {
+				planID := streamResponse.PlanId
+				if passStarted == false {
+					if planID != "" {
+						log.Printf("Receiving messages for Plan ID %v", planID)
+						passStarted = true
+						firstTime, _ := ptypes.Timestamp(message.TimeFirstByteReceived)
+						md.initialTime = firstTime
+						ps.firstByteTime = firstTime
+						ps.initialPlanID = planID
+					}
+
+				}
+				timeFirstByteReceived, _ := ptypes.Timestamp(message.TimeFirstByteReceived)
+				timeLastByteReceived, _ := ptypes.Timestamp(message.TimeLastByteReceived)
+				numBytesInMessage := len(message.Data)
+
+				metric := metric{
+					timeFirstByteReceived: timeFirstByteReceived,
+					timeLastByteReceived:  timeLastByteReceived,
+					dataSize:              numBytesInMessage,
+					metricTime:            time.Now().UTC(),
+				}
+				md.metrics = append(md.metrics, metric)
+				if md.mostRecentTimeLastByteReceived.Before(timeLastByteReceived) {
+					md.mostRecentTimeLastByteReceived = timeLastByteReceived
+				}
+				md.planID = planID
+			}
+
+		case timeOfReporting := <-reportingTicker.C:
+			if passStarted {
+				md.timeOfReporting = timeOfReporting
+				outputDetails := md.compileDetails()
+				if !printedHeader {
+					outputDetails.printHeader(output)
+					printedHeader = true
+				}
+				if (assumeEndAfterDuration > 0) && (outputDetails.totalDataSize == 0) {
+					timeSincePassFirstByte := time.Now().UTC().Sub(ps.firstByteTime)
+					if timeSincePassFirstByte > assumeEndAfterDuration {
+						go func() { passEndedChan <- true }()
+					}
+				} else {
+					ps.totalBytes += outputDetails.totalDataSize
+					ps.totalMbps += outputDetails.dataRateInMbps
+					ps.totalAverageFirstByteLatency += outputDetails.averageFirstByteLatency
+					ps.totalAverageLastByteLatency += outputDetails.averageLastByteLatency
+					ps.numIntervals++
+					ps.lastByteTime = md.mostRecentTimeLastByteReceived
+					ps.totalOutOfOrderData += outputDetails.outOfOrderData
+				}
+
+				if !done && outputDetails.planID != "" {
```

Comment:
I feel there wound't be a case where done is true. If so, you can omit `!done &&`